packages/commonwealth/client/scripts/views/pages/create_community/ProtocolCommunityForm.tsx

```diff
@@ -0,0 +1,399 @@
+              title="Governance Contract"
+              onChangeHandler={(e) =>
+                setSeedWalletMeta({
+                  type: 'multi',
+                  amount: seedWalletMeta['amount'],
+                  address: e,
+                })
+              }
+              value={seedWalletMeta.address}
+            />
+            <InputRow
+              title="Amount of ETH"
+              onChangeHandler={(e) =>
+                setSeedWalletMeta({
+                  type: 'multi',
+                  amount: e,
+                  address: seedWalletMeta['address'],
+                })
+              }
+              value={seedWalletMeta.amount}
+            />
+          </>
+        )}
+      </>
+    );
+  };
+
+  return loading ? (
+    <div className="spinner-group">
+      <CWSpinner />
+      <CWText>Deploying Namespace</CWText>
+    </div>
+  ) : (
+    <div className="CreateCommunityForm">
+      <CWText type="h2" fontWeight="bold">
+        1. General Info
+      </CWText>
+      <CWDivider />
+      <CWText type="caption">
+        Set a name, description, socials and upload an image
+      </CWText>
+      <InputRow
+        title="Name"
+        placeholder="Enter the name of your community"
+        value={name}
+        onChangeHandler={(v) => {
+          setName(v);
+          setId(slugifyPreserveDashes(v));
+        }}
+      />
+      <IdRow id={id} />
+      <InputRow
+        title="Symbol"
+        value={symbol}
+        onChangeHandler={(v) => {
+          setSymbol(v);
+        }}
+      />
+      <CWText type="h2" fontWeight="bold">
+        2. Base Membership
+      </CWText>
+      <CWDivider />
+      <CWText type="caption">
+        Gate access using NFT ownership, ERC20 holders, or anyone on-chain
+      </CWText>
+      <CWDropdown
+        label="Base Chain"
+        options={[{ label: 'ethereum', value: 'ethereum' }]}
+        onSelect={(o) => {
+          setBase(o.value as ChainBase);
+        }}
+      />
+      <CWDropdown
+        label="Membership Access Options"
+        options={membershipOptions}
+        onSelect={(v) => onGateChange(v)}
+        initialValue={membershipOptions[0]}
+      />
+      {gateType !== 'all' && getGateForm(gateType)}
+      <CWText type="h2" fontWeight="bold">
+        3. Seed Your Community Wallet
+      </CWText>
+      <CWDivider />
+      <CWText type="caption">
+        Choose to link an existing Gnosis Safe/Multi-Sig, create a funding
+        proposal, or send from your wallet.
+      </CWText>
+      <CWToggle
+        checked={checked}
+        onChange={() => {
+          setChecked(!checked);
+        }}
+      />
+      {checked && seedWalletForm()}
+      <CWText type="h2" fontWeight="bold">
+        4. Socials and Additional Info
+      </CWText>
+      <CWDivider />
+      {defaultChainRows(chainFormDefaultFields)}
+      <CWButton
+        label="Save changes"
+        disabled={saving || id.length < 1}
+        onClick={async () => {
```

Comment:
can you move this whole function to the function declared above the return

`const handleSaveChanges = () => {} ` 

This way we declutter the visual part of the component and separate out saving logic into the function